parsl/dataflow/dflow.py

```diff
@@ -606,7 +606,18 @@ def sanitize_and_wrap(self, task_id, args, kwargs):
         # Replace item in args
         new_args = []
         for dep in args:
-            if isinstance(dep, Future):
+            if isinstance(dep, list) or isinstance(dep, tuple):
+                new_sub_args = []
+                for sub_dep in dep:
+                    if isinstance(sub_dep, Future):
+                        try:
+                            new_sub_args.extend([sub_dep.result()])
```

Comment:
Is the `sub_dep` future completed at this point for sure? If not, I think this introduces unexpected behaviour (perhaps blocking in the callback handling for other tasks, which can be from arbitrary executor threads, perhaps then deadlocking execution)

There is list of dependencies which the DataFlowKernel waits for, generated by `DataFlowKernel._gather_all_deps` and these changes to sanitize_and_wrap probably need to be reflected there.